### PR TITLE
Fix lint error in /methods and remove usage of internal error package

### PR DESCRIPTION
**Pull request by psheth9**
This diff does following: 

- Remove usage of internal error
- Fix lint errors in `/methods `

### CODE REVIEW

File: cmd/soroban-rpc/internal/methods/get_transactions_test.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func TestGetTransactions_CustomLimitAndCursor(t *testing.T) {
 	assert.Equal(t, toid.New(3, 1, 1).String(), response.Cursor)
 
 	// assert transactions result
-	assert.Equal(t, 3, len(response.Transactions))
+	assert.Len(t, 3, len(response.Transactions))
```
Comment (Shaptic) [inline cmd/soroban-rpc/internal/methods/get_transactions_test.go:182]: You need to drop the len() call here I think.

File: cmd/soroban-rpc/internal/methods/simulate_transaction.go [hunk 2]
```diff
@@ -213,12 +214,16 @@ type SimulateTransactionResponse struct {
...
+	MinResourceFee int64 `json:"minResourceFee,string,omitempty"`
+	// an array of the individual host function call results
+	Results []SimulateHostFunctionResult `json:"results,omitempty"`
+	// the effective cpu and memory cost of the invoked transaction execution.
+	Cost SimulateTransactionCost `json:"cost,omitempty"`
+	// If present, it indicates that a prior RestoreFootprint is required
+	RestorePreamble *RestorePreamble `json:"restorePreamble,omitempty"`
+	// If present, it indicates how the state (ledger entries) will change as a result of the transaction execution.
+	StateChanges []LedgerEntryChange `json:"stateChanges,omitempty"`
+	LatestLedger uint32              `json:"latestLedger"`
```
Comment (Shaptic) [inline cmd/soroban-rpc/internal/methods/simulate_transaction.go:226]: Stared at this forever and can’t see the diff. Halp?

**Review by Shaptic — COMMENTED** [review]
Coupla notes, I think tests will fail cuz of len() but otherwise nice work!!

File: cmd/soroban-rpc/internal/methods/get_transactions_test.go (hunk 1 above)
Comment (psheth9) [reply cmd/soroban-rpc/internal/methods/get_transactions_test.go:182]: ohh shoot right, fast fixes ..

File: cmd/soroban-rpc/internal/methods/simulate_transaction.go (hunk 2 above)
Comment (psheth9) [reply cmd/soroban-rpc/internal/methods/simulate_transaction.go:226]: just formatted comments here to keep it above the variable and to reduce total chars per line :|

**Review by Shaptic — APPROVED** [review]

**Pull request merged**